src/vivarium/framework/resource/interface.py

```diff
@@ -47,6 +47,7 @@ def add_resources(
         component: Component | Manager | None,
         resources: Iterable[str | Resource],
         dependencies: Iterable[str | Resource],
+        are_columns: bool = False,
```

Comment:
Is it better just to have a separate method called add_private_column()? It is only called in one place (i.e. in the PopulationManager), so maybe this is not a very consequential decision, though it does allow us to avoid muddying the waters for all other calls of add_resources(). 

It could look like this:
```python
def add_private_column(component, resources, dependencies):
    columns = [Column(col, component) for col in resources]
    self.add_resources(component, columns, dependencies)
```